src/sql/store.py

```diff
@@ -98,14 +99,22 @@ def __init__(self, store: SQLStore, query: str, with_: Iterable = None):
             )
 
     def __str__(self) -> str:
-        if self._query:
-            with_all = _get_dependencies(self._store, self._with_)
-            parsed_res = parse_one(self._query)
-            for with_key in with_all:
-                with_query = self._store._data[with_key]._query
-                parsed_res = parsed_res.with_(with_key, with_query)
-            return str(parsed_res.sql())
-        return ""
+        with_clause_template = Template(
+            """WITH{% for name in with_ %} "{{name}}" AS ({{saved[name]._query}})\
+{{ "," if not loop.last }}{% endfor %}{{query}}"""
+        )
+        with_clause_template_backtick = Template(
+            """WITH{% for name in with_ %} `{{name}}` AS ({{saved[name]._query}})\
+{{ "," if not loop.last }}{% endfor %}{{query}}"""
+        )
+        is_use_backtick = sql.connection.Connection.is_use_backtick_template()
```

Comment:
For readability and to facilitate future changes, I think we can add some comments here. Also, I see there is no difference between the 2 templates besides the {{name}} wrappers. Can we use a string formatted here?